### PR TITLE
initialize error_code to 0

### PR DESCRIPTION
**Pull request by forwardpointer**
### Type

- [ ] Bug fix
- [ ] Feature addition
- [ ] Feature update
- [ ] Breaking change
- [ ] Documentation
- [X] Infrastructure

### Goals

Initialize loop->error_code to 0 within event_loop_initialize() to prevent this being a bug if event_loop_stop() is no longer the unique place where loop->terminate_pending is set.

### Technical Details

None.

### Test Results

None.

### Reviewer Focus

None.

### Future Work

None.

### Checklist

- [X] Build target `all` compiles cleanly.
- [ ] cppcheck produces no output.
- [ ] clang-format delta produced no new output.
- [ ] Newly added functions include doxygen-style comment block.

### CODE REVIEW

**Comment by forwardpointer** [comment]
Sorry, I'm not sure why this PR includes commits that were already merged with main in the previous PR. I'll try to remove them somehow.

File: src/utils/event_loop.c [hunk 1]
```diff
@@ -268,6 +268,8 @@ event_loop_initialize(struct event_loop *loop)
     int ret = event_loop_epoll_initialize(loop);
     if (ret < 0)
         return ret;
+    
+    loop->exit_code = 0;
```
Comment (abeltrano) [inline src/utils/event_loop.c:272]: Should this be done in `event_loop_epoll_initialize` instead?

**Review by abeltrano — APPROVED** [review]

**Comment by abeltrano** [comment]
> Sorry, I'm not sure why this PR includes commits that were already merged with main in the previous PR. I'll try to remove them somehow.

You can rebase to main then force push to fix that.

File: src/utils/event_loop.c (hunk 1 above)
Comment (forwardpointer) [reply src/utils/event_loop.c:272]: is the exit_code related to epoll?

Comment (abeltrano) [reply src/utils/event_loop.c:272]: Oops sorry, no, you're right, this is the right place.

**Comment by abeltrano** [comment]
Make sure to set your name and e-mail in git before pushing commits:

```shell
git config --global user.name "Shane Guan"
git config --global user.email "<youremail>@microsoft.com"
```

**Pull request merged**